SOURCE/Main_DOODZ.c

```diff
@@ -144,243 +132,198 @@ int RunMDOODZ( int nargs, char *args[] ) {
-            SetBCs_new( &mesh, &model, scaling, &particles, &materials);
+        // Initialise particle fields
+        PartInit( &particles, &model );
+        
+        // Initial grid tags
+        SetBCs( &mesh, &model, scaling , &particles, &materials, &topo);
+        if ( model.free_surf == 1 ) {
             
-            LoadIniParticles( PartFileName, &particles, &mesh, &topo_chain, &topo_chain_ini, &model, scaling );
+            // Define the horizontal position of the surface marker chain
+            SetTopoChainHorizontalCoords( &topo,     &topo_chain,     model, mesh, scaling );
+            SetTopoChainHorizontalCoords( &topo_ini, &topo_chain_ini, model, mesh, scaling );
             
-            if ( model.free_surf == 1 ) {
-                // Project topography on vertices
-                ProjectTopography( &topo, &topo_chain, model, mesh, scaling, mesh.xg_coord, 0 );
-                
-                // Marker chain polynomial fit
-                MarkerChainPolyFit( &topo, &topo_chain, model, mesh );
-                
-                // Call cell flagging routine for free surface calculations
-                CellFlagging( &mesh, model, topo, scaling );
-            }
+            // Define the vertical position of the surface marker chain
+            BuildInitialTopography( &topo,     &topo_chain,     model, mesh, scaling );
+            BuildInitialTopography( &topo_ini, &topo_chain_ini, model, mesh, scaling );
             
-#else
-            // Initial grid tags
-            SetBCs( &mesh, &model, scaling , &particles, &materials, &topo);
-            if ( model.free_surf == 1 ) {
-                
-                // Define the horizontal position of the surface marker chain
-                SetTopoChainHorizontalCoords( &topo,     &topo_chain,     model, mesh, scaling );
-                SetTopoChainHorizontalCoords( &topo_ini, &topo_chain_ini, model, mesh, scaling );
-                
-                // Define the vertical position of the surface marker chain
-                BuildInitialTopography( &topo,     &topo_chain,     model, mesh, scaling );
-                BuildInitialTopography( &topo_ini, &topo_chain_ini, model, mesh, scaling );
-                
-                // Project topography on vertices
-                ProjectTopography( &topo, &topo_chain, model, mesh, scaling, mesh.xg_coord, 0 );
-                
-                // Marker chain polynomial fit
-                MarkerChainPolyFit( &topo, &topo_chain, model, mesh );
-                
-                // Call cell flagging routine for free surface calculations
-                CellFlagging( &mesh, model, topo, scaling );
-            }
-            // Set particles coordinates
-            PutPartInBox( &particles, &mesh, model, topo, scaling );
+            // Project topography on vertices
+            ProjectTopography( &topo, &topo_chain, model, mesh, scaling, mesh.xg_coord, 0 );
```

Comment:
What's the difference between "ProjectTopography" and "BuildInitialTopography" methods?
Maybe we should reflect it somehow in name?

Reply:
"ProjectTopography" projects topography from the markers onto the grid.
"BuildInitialTopography" builds the initial topography (on the markers)
I've left a TODO for future refactoring.